scalyr_agent/tests/log_processing_test.py

```diff
@@ -2031,9 +2031,10 @@ def test_matches_recursive_glob(self):
         if sys.version_info >= (2, 6):
             matcher = LogMatcher(self.__config, self._create_log_config(self.__glob_recursive))
             processors = matcher.find_matches(dict(), dict())
+            processors.sort() # required to make this test deterministic.
```

Comment:
Hmm.. I would suggest we do this slightly differently.  This is doing the sorting at the test-level and fixing it for this specific test.  Let's make the `find_matches` call itself deterministic.  Most likely, this means you need to sort the output of whatever is doing the globbing.  That way future test cases won't trip over this as well.  Also, sometimes the order files are listed matter to the agent.. so making everything deterministic can avoid weird bugs in the future as well.

Reply:
Makes sense.
Is sorting the result just above the return here what you are thinking?https://github.com/scalyr/scalyr-agent-2/blob/bchristie/fix-flaky-glob2-tests/scalyr_agent/log_processing.py#L2559

Reply:
I was actually thinking of sorting the results from `glob.glob` here https://github.com/scalyr/scalyr-agent-2/blob/bchristie/fix-flaky-glob2-tests/scalyr_agent/log_processing.py#L2489